docs/manual/DEVELOPER_MANUAL.md

```diff
@@ -0,0 +1,892 @@
+5. [Commands](#commands)
+    1. [kore-exec](#commands-kore-exec)
+    2. [kore-repl](#commands-kore-repl)
+6. [K Framework](#kframework)
+7. [Application Notes](#application-notes)
+8. [Glossary](#glossary)
+
+## [Introduction]{#introduction}
+
+### [Motivation]{#introduction-motivation}
+
+### [Document Structure]{#introduction-document-structure}
+
+## [Basics]{#basics}
+
+## [Design]{#design}
+
+## [Implementation]{#implementation}
+
+### [Context Map]{#implementation-context-map}
+
+> [Domain Driven Design] deals with large models by dividing them into different Bounded Contexts and being explicit about their interrelationships."
+
+[BoundedContext - martinfowler.com](https://martinfowler.com/bliki/BoundedContext.html)
+
+Our boundaries are not very strict, because our domain is well-defined; that is, Kore is given a formal semantics. The map has basically two kinds of entities:
+
+* *Languages* define an interface between contexts. For us, these are mostly internal representations of Kore patterns and definitions.
+* *Workflows* implement the behavior of the model, for example: parsing, validation, rewriting, simplification, proving.
+
+Sometimes *languages* and *workflows* overlap, but this should be avoided.
+
+#### [Low-level]{#implementation-context-map-low-level}
+
+
+![context-map-low-level.jpg](./img/implementation/context-map/context-map-low-level.jpg)
+
+
+This diagram is intended to be normative. Question carefully any instance where the actual dependencies differ significantly from the relationships shown here. Note that all internal modules depend on Kore.Internal, but these are omitted for clarity. The diagram only depicts the core domain models. There are utility contexts for logging, databases, etc. which are not shown.
+
+#### [Interface]{#implementation-context-map-interface}
+
+
+![context-map-interface.jpg](./img/implementation/context-map/context-map-interface.jpg)
+
+
+#### [Workflow]{#implementation-context-map-workflow}
+
+
+![context-map-workflow.jpg](./img/implementation/context-map/context-map-workflow.jpg)
+
+
+Kore.Exec and Kore.Repl control the input/output cycle with the user. These modules drive the data flow through the backend. Input passes through parsing and validation. The interface between these is mediated by Kore.Syntax. The valid definition and claims or patterns pass to Kore.Reachability, with the interface mediated by Kore.Internal and the other representations. The claims or program iterate through Kore.Rewrite and Kore.Simplify according to the controller, Kore.Reachability.
+
+### [Hooks]{#implementation-hooks}
+
+[Pull request #632](https://github.com/kframework/kore/pull/632/files) is a self-contained example of adding a builtin symbol. When adding a builtin symbol we should add:
+
+1. An entry to the hooks document.
+2. A symbol verifier.
+3. An evaluator for concrete arguments (at least).
+4. A declaration to `Test.Kore.Builtin.Definition`.
+5. A few tests with different concrete arguments.
```

Comment:
Let's keep this for now, but remember it's probably very out of date (not necessarily the method itself, but the PR we are linking to). We should probably add a TODO here or something to remind us.

Reply:
Added TODO with link to your comment, also made this information it's own section for easy access via table of contents.